src/test/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStepWorkflowTest.java

```diff
@@ -282,4 +283,24 @@ public void sshAgentDocker() throws Exception {
         });
     }
 
+    @Issue("JENKINS-32104")
+    @Test
+    public void testMissingCredential() {
+        assumeFalse(Functions.isWindows());
+
+        story.addStep(new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                WorkflowJob job = story.j.jenkins.createProject(WorkflowJob.class, "sshAgentAvailable");
+                job.setDefinition(new CpsFlowDefinition(""
+                  + "node {\n"
+                  + "  sshagent (credentials: ['nonexistent']) {\n"
+                  + "  }\n"
+                  + "}\n", true)
+                );
+                story.j.assertBuildStatus(Result.FAILURE, job.scheduleBuild2(0));
```

Comment:
For “extra credit”, use `assertLogContains` to check for the expected error message. (Otherwise the test could pass for all sorts of incorrect reasons, such as a typo in the Pipeline script resulting in a Groovy syntax error!)